.github/workflows/scripts/generate-new-contributors.js

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+import { simpleGit } from 'simple-git';
+const git = simpleGit();
+
+const REPO_NAME = "opentelemetry-collector-contrib"
+const REPO_OWNER = "open-telemetry"
+
+const firstTimeContributorText = "We are thrilled to welcome our new first-time contributors to this project. Thank you for your contributions "
+
+function getUniqueCombinations(data) {
+    const uniqueSet = new Set();
+    const uniqueArray = [];
+
+    data.forEach(item => {
+        const combination = `${item.name}-${item.email}`;
+        if (!uniqueSet.has(combination)) {
+            uniqueSet.add(combination);
+            uniqueArray.push(item);
+        }
+    });
+
+    return uniqueArray;
+}
+
+async function getFirstTimeContributors(fromTag, toTag) {
+    try {
+        const fromCommit = await git.revparse(fromTag)
+        const toCommit = await git.revparse(toTag)
+        // Get the list of commits between two tags
+        const releaseCommits = await git.log({
+            from: fromCommit, to: toCommit,
+            format: "oneline"
+        }).then(result => result.all);
+
+        const authorsAndHashes = releaseCommits.map(commit => {
+            return {name: commit.author_name, email: commit.author_email, hash: commit.hash}
+        });
+
+        const uniqueAuthorsAndHashes = getUniqueCombinations(authorsAndHashes);
+        const allCommits = await git.log({
+            format: "oneline",
+        })
+
+        const firstTimeContributorsAndHashes = [];
```

Comment:
Should we take care to filter bots?

Reply:
you mean by filtering for `[bot]` in the username? i don't think it's really worth the effort. how often does it happen that a new bot is introduced? also it doesn't really hurt if it gets a shout-out 😄